### PR TITLE
Update AMSU-A YAML files for UFO evaluation and End-to-End configuration YAML files 

### PR DESCRIPTION
**Pull request by emilyhcliu**
Updates for UFO evaluation due to crtm upgrade from v2.3.0 to v2.4.1
- AMSU-A YAML files under testing
- Input obs and geoval files on Orion and Hera
- UFO evaluation (compared to GSI) result summary:
   - AMSU-A N19:
      - HofX: 100%
      - Obs Count passed QC: 100%
      - Obs Error: ~100% four observations differences at the edge of sea/sea ice around south pole
   - AMSU-A N18:
     - HofX: 100%
     - Obs Count passed QC: 100%
     - Obs Error: ~100% except for one observation located at  lat: 77.7  lon: 377.7
  - AMSU-A N15: 
    - HofX: 100%
    - Obs count passed QC: 100%
    -  Obs Error: 100%

  - AMSU-A Metop-A:
    - HofX: 100%
    - Obs Count passed QC: 100%
    - Obs Error: ~ 100% (three obs difference ~ 0.02K)
   
  - AMSU-A MetOp-B
    - HofX: 100%
    - Obs Count passed QC: 100%
    - Obs Error: 100%

  - AMSU-A MetOp-C:
    - HofX: 100%
    -  ObsCount passed QC: 100%
    -  Obs Error: ~100% (one obs difference)
 Please check the validation plots in the comment below.

Updates for End-to-End configuration
- Update AMSU-A N19
- Add AMSU-A N18, N15, Metop-A, Metop-B, and Metop-C

Notes:
1. There is a [paring PR](https://github.com/JCSDA-internal/ufo/pull/3094)  in JCSDA UFO repository
    It is OK to merge this before the paring UFO PR 
    The related CI tests will pass once the paring  UFO PR is merged in to UFO develop.

2. @xincjin-NOAA has a [draft PR ](https://github.com/NOAA-EMC/GDASApp/pull/754)for the BUFR to IODA converter for AMSU-A 
3. @xincjin-NOAA  and I will work on the AMSU-A end-to-end test after BUFR converter for AMSU-A is merged.

### CODE REVIEW

**Comment by emilyhcliu** [comment]
### UFO evaluation results (using AMSU-A MetOp-C as an example)

**HofX Difference (JEDI vs. GSI)**

HofX_Difference_amsua_metop-c_diag_2021080100_amsua_metop-c

**Final Obs Error Difference (JEDI vs. GSI)**
**The difference is from one sounding location at 78N and 157 W.**  

ObsErr_Difference_amsua_metop-c_diag_2021080100_amsua_metop-c

**Difference of Data Count Passed QC between JEDI and GSI**

QCCount_Diff_amsua_metop-c_diag_2021080100_amsua_metop-c

File: parm/atm/obs/config/amsua_metop-a.yaml [hunk 1]
```diff
@@ -0,0 +1,484 @@
+obs space:
+  name: amsua_metop-a
+  obsdatain:
+    engine:
+      type: H5File
+      obsfile: $(DATA)/obs/$(OPREFIX)amsua_metop-a.{{ current_cycle | to_YMDH }}.nc4
```
Comment (RussTreadon-NOAA) [inline parm/atm/obs/config/amsua_metop-a.yaml]: prepatmiodaobs creates ioda format dump files using a different naming convention.  prepatmiodaobs will create `$(OPREFIX)amsua_metop-a.tm00.nc` for amsua_metop-a.  

The same naming convention comment applies to the dump `obsfile` in other amsua yamls in this PR.

File: parm/atm/obs/config/amsua_metop-a.yaml [hunk 2]
```diff
@@ -0,0 +1,484 @@
...
+- filter: Variable Assignment
+  assignments:
+  - name: DerivedMetaData/ObsErrorFactorTopo
+    channels: *amsua_metop-a_channels
+    type: float
+    function:
+      name: ObsFunction/ObsErrorFactorTopoRad
+      channels: *amsua_metop-a_channels
+      options:
+        sensor: *Sensor_ID
```
Comment (RussTreadon-NOAA) [inline parm/atm/obs/config/amsua_metop-a.yaml:333]: alias `*Sensor_ID` is referenced here and elsewhere, but `&Sensor_ID` is not defined in this yaml file.   

Job gdasatmanlinit failed when processing this yaml with the error

```
yaml.composer.ComposerError: found undefined alias 'Sensor_ID'
  in "/work2/noaa/da/rtreadon/gdas-validation/global-workflow/sorc/gdas.cd/parm/atm/obs/config//amsua_metop-a.yaml", line 312, column 17
```

This comment applies to other amsua yamls in this PR.

File: parm/atm/obs/testing/amsua_metop-a.yaml [hunk 3]
```diff
@@ -18,7 +18,7 @@ obs space:
         type: H5File
         obsfile: !ENV amsua_metop-a_diag_${CDATE}.nc4
```
Comment (RussTreadon-NOAA) [inline parm/atm/obs/testing/amsua_metop-a.yaml:10]: I don't think `!ENV` is the way we now specify `obsfile`.   `config/amsua_metop-a.yaml` specifies the ability `obsfile` as `$(DATA)/obs/$(OPREFIX)amsua_metop-a.`   This comment applies to other amsua yamls in `testing`

There are other occurrences of `!ENV` in the testing amsua yamls.   Do we need to change these?

**Review by RussTreadon-NOAA — COMMENTED** [review]
This review is a spot check of yamls.  Unable to fully test yamls in g-w because all pieces (converters) are not available in this PR or `gdas-validation`.

File: parm/atm/obs/config/amsua_metop-a.yaml (hunk 1 above)
Comment (emilyhcliu) [reply parm/atm/obs/config/amsua_metop-a.yaml]: Got it.  The convention used above is old.  I will update it
We have atms_npp and atms_n20 validated in the global workflow. I will modify AMSU-A to be consistent with the ATMS.

File: parm/atm/obs/testing/amsua_metop-a.yaml (hunk 3 above)
Comment (emilyhcliu) [reply parm/atm/obs/testing/amsua_metop-a.yaml:10]: @RussTreadon-NOAA @CoryMartin-NOAA 
I fixed the following:
1.  Modified obsfile name to follow the naming convention
2. defined Sensor_ID

**Comment by RussTreadon-NOAA** [comment]
Why is this PR, #763, separate from #754.   It seems that we need both (along with an iodaconv update) to run g-w tests.   Is this correct?

**Comment by emilyhcliu** [comment]
> Why is this PR, #763, separate from #754. It seems that we need both (along with an iodaconv update) to run g-w tests. Is this correct?

Yes, we need both PRs to run g-w test. 
PR #754 is for AMSU-A BUFR converter.  It is using a hybrid of YAML and Python, a new feature recently developed in BUFR converter.  We are using AMSU-A to test this new feature.  

So, the goal for AMSU-A in this code sprint is:
1.  Checking in the updated YAML files for testing and config (PR 763)
2. Checking in the required components for converting BUFR to IODA with the hybrid method (PR 754)

Testing AMSU-A in g-w will be done after the above is ready after the code sprint.

**Comment by emilyhcliu** [comment]
@RussTreadon-NOAA 
I am adding two additional filters in the YAML.
I will update the YAMLs pretty soon (in 30 minutes)

**Comment by RussTreadon-NOAA** [comment]
@emilyhcliu , OK.  Let me pause my examination of PR #763 and #754.   You are still making changes in PR #763.  PR #754 is still in draft mode.   When both are finalized, I will merge them together in my local working copy and then attempt to exercise them from g-w.

Before finalizing both PRs, it would be helpful to resync them with the current head of `feature/gdas-validation`.

**Comment by emilyhcliu** [comment]
@RussTreadon-NOAA I have completed the modification for AMSU-A YAMLs and also updated the branch with the head of feature/gdas-validation.

File: CMakeLists.txt [hunk 4]
```diff
@@ -80,7 +80,8 @@ if(BUILD_GDASBUNDLE)
   option(ENABLE_IODA_DATA "Obtain ioda test data from ioda-data repository (vs tarball)" ON)
   ecbuild_bundle( PROJECT ioda  GIT "https://github.com/jcsda/ioda.git" BRANCH develop  )
   option(ENABLE_UFO_DATA "Obtain ufo test data from ufo-data repository (vs tarball)" ON)
-  ecbuild_bundle( PROJECT ufo   GIT "https://github.com/jcsda/ufo.git" BRANCH develop )
```
Comment (CoryMartin-NOAA) [inline CMakeLists.txt]: we should revert this change

File: parm/atm/obs/testing/amsua_metop-a.yaml (hunk 3 above)
Comment (CoryMartin-NOAA) [reply parm/atm/obs/testing/amsua_metop-a.yaml:10]: I think it's okay for the testing YAMLs to include this old YAML style, especially if others in testing look like this, they are intended to be only for UFO acceptance and not "real" end to end use.

File: parm/atm/variational/3dvar_dripcg.yaml [hunk 5]
```diff
@@ -33,7 +33,7 @@ variational:
   minimizer:
     algorithm: DRIPCG
   iterations:
-  - ninner: 2
```
Comment (CoryMartin-NOAA) [inline parm/atm/variational/3dvar_dripcg.yaml]: we should revert the changes to this file, it shouldn't change in the PR.

Comment (emilyhcliu) [reply parm/atm/variational/3dvar_dripcg.yaml]: @CoryMartin-NOAA I accidentally commited this in.  Try a few things but I could not revert this file to the base (feature/gdas-validation) successfully.  Could you help?

Comment (emilyhcliu) [reply parm/atm/variational/3dvar_dripcg.yaml]: @CoryMartin-NOAA 
I will do a minor re-arrange of these AMSU-A YAMLS.  The re-arrangement will not change results.  It is just to make the YAML more readable.  I will have the re-arrangement done this evening.

**Comment by emilyhcliu** [comment]
I am done updating the YAML files!  
But, need help to revert `dvar_dripcg.yaml` to the original.

File: parm/atm/variational/3dvar_dripcg.yaml (hunk 5 above)
Comment (CoryMartin-NOAA) [reply parm/atm/variational/3dvar_dripcg.yaml]: @emilyhcliu I reverted, FYI I used:
`git checkout feature/gdas-validation -- parm/atm/variational/3dvar_dripcg.yaml`

Comment (emilyhcliu) [reply parm/atm/variational/3dvar_dripcg.yaml]: Supre!! Thank you @CoryMartin-NOAA

**Comment by RussTreadon-NOAA** [comment]
I'm a bit confused.  Does this PR, #763, need to be merged _before_ or _concurrent_ with PR #754?

**Comment by CoryMartin-NOAA** [comment]
@RussTreadon-NOAA probably either concurrent with or after #754

**Comment by RussTreadon-NOAA** [comment]
Thank you @CoryMartin-NOAA .  I'll stand down on testing this PR and wait until #754 is ready for review.

**Comment by emilyhcliu** [comment]
@xincjin-NOAA combined the updates in this PR to his working branch.  So, I am closing this PR.
